
Rediffed hotfix from X.Org Bugzilla to prevent X from haning during startup
on e.g. PowerPC Apple {i,Power}Books ...

  - Rene Rebe <rene@rocklinux.org>

--- xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	2004-06-16 11:43:59.000000000 +0200
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	2004-08-19 13:15:37.473875024 +0200
@@ -1634,8 +1634,8 @@
 				info->Flags = 0;
 				break;
 			    }
-			    tmp_mode = tmp_mode->next;
 			}
+			tmp_mode = tmp_mode->next;
 		    }
 		    if ((info->DotClock == 0) && !pRADEONEnt->MonInfo1) {
 			xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
